### PR TITLE
feat(cell): model header sizes for messages transmitted by cell

### PR DESCRIPTION
**Pull request by kschrab**
## Description

Introduces a simplified modelling of headers added to payload of messages sent trough the cellular stack.

* Adding a 20-byte sized TCP or 8-byte sized UDP header to each message, according to its transport mode.
* Always adding a 20-byte sized IP header to each message.
* When adding the message to the cellular part of the network (e.g. sent from a vehicle), additional bytes are added to represent the headers of the cellular stack. Since the actual cellular stack may change and depend on the simulation scenario, we named it `cellularHeader` (and is configurable). In the default case, we assume an LTE stack adding another 26 bytes to each message (PDCP (6 bytes) + RLC (4 bytes) + MAC (10 bytes) + PHY (6 bytes) ).
* When the receiver or sender is a server, it is assumed that it is connected directly to the core network and therefore, an Ethernet header is added. Again, the technology used here can vary, thus the size can be configured and is set to 38 bytes in the default case.
* Any additional headers which may be added within the transmission trough the network are not considered, as this is out of scope of the simplified model `mosaic-cell` provides.
* All header sizes can be configured in the `cell_config.json`:

```json
{
    "networkConfigurationFile": "network.json",
    "regionConfigurationFile": "regions.json",
    "headerLengths": {
        "udpHeader": "8 Bytes",
        "tcpHeader": "20 Bytes",
        "ipHeader": "20 Bytes",
        "cellularHeader": "26 Bytes",
        "ethernetHeader": "38 Bytes"
     }
}
```

Open issues:

* [x] Fix tests
* [ ] Fix documentation

## Issue(s) related to this PR

* Resolves internal issue 567
  
## Affected parts of the online documentation

Changes in the documentation required?

Yes, must be done.

## Definition of Done

**Prerequisites**

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] Your GitHub user id is linked with your Eclipse Account.

**Required**

- [x] The title of this merge request follows the scheme `type(scope): description`  (in the style of [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary))
- [x] You have assigned a suitable label to this pull request (e.g., `enhancement`, or `bugfix`)
- [x] `origin/main` has been merged into your Fork.
- [x] Coding guidelines have been followed (see CONTRIBUTING.md).
- [x] All checks on GitHub pass.
- [x] All tests on [Jenkins](https://ci.eclipse.org/mosaic/job/mosaic/) pass.

**Requested** (can be enforced by maintainers)

- [x] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [ ] If a bug has been fixed, a new unit test has been written (beforehand) to prove misbehavior
- [x] There are no new SpotBugs warnings.

### CODE REVIEW

File: fed/mosaic-cell/src/main/java/org/eclipse/mosaic/fed/cell/utility/CapacityUtility.java [hunk 1]
```diff
@@ -133,8 +138,28 @@ public static void consumeCapacity(TransmissionMode mode, CNetworkProperties reg
...
+    public static long getMessageLengthWithHeaders(V2xMessage msg, String senderOrReceiver) {
+        final CCell.CHeaderLengths headerLengths = ConfigurationData.INSTANCE.getCellConfig().headerLengths;
+        final long linkLayerHeader;
+        if (senderOrReceiver != null && (isServer(senderOrReceiver) || isTmc(senderOrReceiver))) {
+            // let's assume everything is connected via cellular link, except servers and tmcs which are connected with the backbone
+            linkLayerHeader = headerLengths.ethernetHeader;
+        } else {
+            linkLayerHeader = headerLengths.cellularHeader;
+        }
+        switch (msg.getRouting().getDestination().getProtocolType()) {
```
[the remark this answers is not in the document]
Comment (kschrab) [reply fed/mosaic-cell/src/main/java/org/eclipse/mosaic/fed/cell/utility/CapacityUtility.java:150]: > However, if users may need more freedom in definition of their headers,

They do have complete control over the header sizes already, since they can override all the proposed values in the config. I included the header sizes of the PHY layer to `ethernetHeader`/`cellularHeader` since they differ between both technologies. Moving PHY sizes to a more general `additionalOverhead` field would use the same header size for both cellular/link-based transmission. I see your point, that PHY depends on the infrastructure being used (optical vs. wires), but it is already configurable by just using different values here.

tl;dr, We could arguably just ignore PHY completely if you wish to. Still, I cannot see any value by adding the "additionalOverhead" field since the user can already configure additional header sizes (with more control) by re-configuring/overriding the proposed fields.
